lib/gc/Transforms/TilingUsingInterfaceX.cpp

```diff
@@ -255,6 +256,28 @@ SmallVector<LoopLikeOpInterface> mlir::scfX::getOuterNestLoopsWhile(
   return {nestLoops.rbegin(), nestLoops.rend()};
 }
 
+/// A listener that watches which ops were erased.
+struct ErasedOpListener : public RewriterBase::Listener {
+private:
+  /// Pointers to all erased operations and blocks.
+  DenseSet<void *> erased;
+  // Hook old listener.
+  OpBuilder::Listener *oldListenerHook = nullptr;
+
+public:
+  ErasedOpListener() = default;
+  ErasedOpListener(OpBuilder::Listener *oldListener)
+      : oldListenerHook(oldListener) {}
+  void notifyOperationErased(Operation *op) override {
+    // Call old listener hook.
+    if (auto *oldListener =
+            dyn_cast_if_present<RewriterBase::Listener>(oldListenerHook))
```

Comment:
why do we need this cast?

Reply:
`notifyOperationErased` is just the member function of `RewriterBase::Listener`, which is the sub-class of `OpBuilder::Listener`.

Reply:
Offline discussed.